packages/expo-constants/src/Constants.ts

```diff
@@ -35,6 +37,18 @@ if (ExponentConstants && ExponentConstants.manifest) {
     manifest = JSON.parse(manifest);
   }
 }
+// If expo-updates defines a non-empty manifest, use that one instead
```

Comment:
we should move `if (ExponentConstants && ExponentConstants.manifest) {` below the `if (NativeModulesProxy.ExpoUpdates) {` block and change the conditional to:

`if (ExponentConstants && ExponentConstants.manifest && !manifest) {`

this way we skip parsing the embedded manifest if we end up using the manifest from updates